cilium-cli/install/helm.go

```diff
@@ -64,8 +64,12 @@ func (k *K8sInstaller) getHelmValues() (map[string]interface{}, error) {
 				// Can be removed once we drop support for <1.14.0
 				helmMapOpts["tunnel"] = tunnelDisabled
 			}
-			// TODO(tgraf) Is this really sane?
+			// AL2023 uses ens interfaces, but we default to eth interfaces for everything else for backwards compatibility,
+			// since the CLI was assuming eth interfaces before support for AL2023 was introduced
 			helmMapOpts["egressMasqueradeInterfaces"] = "eth0"
+			if k.params.AWS.AwsNodeImageFamily == AwsNodeImageFamilyAmazonLinux2023 {
+				helmMapOpts["egressMasqueradeInterfaces"] = "ens5"
```

Comment:
How certain are we that `ens5` is the correct interface on all nodes? Should we use a prefix match such as `ens+` here? I believe even the previous `eth0` probably should have been `eth+` to be honest, i.e. pod-to-pod traffic on secondary interfaces probably should also be subject to masquerading. 

Unfortunately our docs are a bit conflicting here. The ENI docs do recommend `eth+`, where as the Helm getting started guide just sets `eth0`:

https://docs.cilium.io/en/v1.16/network/concepts/routing/#id6
https://docs.cilium.io/en/latest/installation/k8s-install-helm/#k8s-install-helm